File: snake_game.py
import pygame
import random

# Initialize Pygame
pygame.init()

# Constants
WIDTH, HEIGHT = 800, 600
CELL_SIZE = 20
CELL_WIDTH, CELL_HEIGHT = WIDTH // CELL_SIZE, HEIGHT // CELL_SIZE
FPS = 10

# Colors
WHITE = (255, 255, 255)
RED = (255, 0, 0)
GREEN = (0, 255, 0)
BLACK = (0, 0, 0)

# Set up the display
screen = pygame.display.set_mode((WIDTH, HEIGHT))
pygame.display.set_caption('Pythonic Pursuit')

clock = pygame.time.Clock()

# Snake initial position and direction
snake = [(CELL_WIDTH // 2, CELL_HEIGHT // 2)]
snake_direction = (0, 0)
snake_speed = 1

# Fruit initial position
fruit = (random.randint(0, CELL_WIDTH - 1), random.randint(0, CELL_HEIGHT - 1))


def draw_snake():
    for segment in snake:
        pygame.draw.rect(screen, GREEN, (segment[0] * CELL_SIZE, segment[1] * CELL_SIZE, CELL_SIZE, CELL_SIZE))


def draw_fruit():
    pygame.draw.rect(screen, RED, (fruit[0] * CELL_SIZE, fruit[1] * CELL_SIZE, CELL_SIZE, CELL_SIZE))


running = True
while running:
    screen.fill(BLACK)

    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            running = False
        elif event.type == pygame.KEYDOWN:
            if event.key == pygame.K_UP:
                snake_direction = (0, -1)
            elif event.key == pygame.K_DOWN:
                snake_direction = (0, 1)
            elif event.key == pygame.K_LEFT:
                snake_direction = (-1, 0)
            elif event.key == pygame.K_RIGHT:
                snake_direction = (1, 0)

    # Update snake position
    snake_head = (snake[-1][0] + snake_direction[0], snake[-1][1] + snake_direction[1])
    snake.append(snake_head)
    if snake_head == fruit:
        fruit = (random.randint(0, CELL_WIDTH - 1), random.randint(0, CELL_HEIGHT - 1))
    else:
        snake.pop(0)

    # Draw elements
    draw_fruit()
    draw_snake()

    # Update display
    pygame.display.update()
    clock.tick(FPS)

pygame.quit()

File: snake_game.js
<!DOCTYPE html>
<html>
<head>
  <title>Snake Game</title>
  <style>
    canvas {
      border: 1px solid black;
      display: block;
      margin: 0 auto;
    }
  </style>
</head>
<body>
  <canvas id="gameCanvas" width="400" height="400"></canvas>
  <script>
    const canvas = document.getElementById('gameCanvas');
    const ctx = canvas.getContext('2d');

    const cellSize = 20;
    const cellNumber = 20;

    let snake = [{ x: cellNumber / 2, y: cellNumber / 2 }];
    let direction = { x: 0, y: 0 };
    let speed = 5;
    let fruit = { x: Math.floor(Math.random() * cellNumber), y: Math.floor(Math.random() * cellNumber) };

    function draw() {
      ctx.clearRect(0, 0, canvas.width, canvas.height);

      // Draw snake
      ctx.fillStyle = 'green';
      for (let i = 0; i < snake.length; i++) {
        ctx.fillRect(snake[i].x * cellSize, snake[i].y * cellSize, cellSize, cellSize);
      }

      // Draw fruit
      ctx.fillStyle = 'red';
      ctx.fillRect(fruit.x * cellSize, fruit.y * cellSize, cellSize, cellSize);
    }

    function update() {
      // Update snake position
      const head = { x: snake[snake.length - 1].x + direction.x, y: snake[snake.length - 1].y + direction.y };
      if (head.x < 0 || head.x >= cellNumber || head.y < 0 || head.y >= cellNumber || snake.some(seg => seg.x === head.x && seg.y === head.y)) {
        alert('Game Over!');
        location.reload();
      }

      snake.push(head);
      if (head.x === fruit.x && head.y === fruit.y) {
        fruit = { x: Math.floor(Math.random() * cellNumber), y: Math.floor(Math.random() * cellNumber) };
      } else {
        snake.shift();
      }
    }

    function gameLoop() {
      update();
      draw();
      setTimeout(gameLoop, 1000 / speed);
    }

    document.addEventListener('keydown', (event) => {
      switch (event.key) {
        case 'ArrowUp':
          direction = { x: 0, y: -1 };
          break;
        case 'ArrowDown':
          direction = { x: 0, y: 1 };
          break;
        case 'ArrowLeft':
          direction = { x: -1, y: 0 };
          break;
        case 'ArrowRight':
          direction = { x: 1, y: 0 };
          break;
      }
    });

    gameLoop();
  </script>
</body>
</html>
